src/Tizen.NUI/src/internal/Layouting/RelativeLayout.cs

```diff
@@ -0,0 +1,280 @@
+/* Copyright (c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Tizen.NUI.BaseComponents;
+using Tizen.NUI.Binding;
+using Tizen.NUI.Xaml;
+
+namespace Tizen.NUI
+{
+    // Methods and Properties for calculation.
+    public partial class RelativeLayout
+    {
+        readonly static Dictionary<View, Relative> HorizontalRelativeCache = new Dictionary<View, Relative>();
+        readonly static Dictionary<View, Relative> VerticalRelativeCache = new Dictionary<View, Relative>();
+        private static readonly RelativePropertyGetters RelativeHorizontalPropertyGetters = new RelativePropertyGetters
+        {
+            RelativeCache = HorizontalRelativeCache,
+            GetTargetFrom = GetLeftTarget,
+            GetTargetTo = GetRightTarget,
+            GetRelativeOffsetFrom = GetLeftRelativeOffset,
+            GetRelativeOffsetTo = GetRightRelativeOffset,
+            GetAlignment = GetHorizontalAlignment,
+            GetFill = GetFillHorizontal,
+            GetMeasuredSize = (item) => { return item.MeasuredWidth.Size.AsDecimal(); }
+        };
+        private static readonly RelativePropertyGetters RelativeVerticalPropertyGetters = new RelativePropertyGetters
+        {
+            RelativeCache = VerticalRelativeCache,
+            GetTargetFrom = GetTopTarget,
+            GetTargetTo = GetBottomTarget,
+            GetRelativeOffsetFrom = GetTopRelativeOffset,
+            GetRelativeOffsetTo = GetBottomRelativeOffset,
+            GetAlignment = GetVerticalAlignment,
+            GetFill = GetFillVertical,
+            GetMeasuredSize = (item) => { return item.MeasuredHeight.Size.AsDecimal(); }
+        };
+
+        delegate Relative GetRelative(View view, in float layoutSize);
+
+        private bool ValidateRelative(Relative relative, in float layoutSize, bool isFixedRelative)
+        {
+            float spaceSize = relative.spaceGeometry.Size;
+            float viewPosion = relative.viewGeometry.Position;
```

Comment:
is it typo? (viewPosion -> viewPosition)

Reply:
typo has been fixed! thank you~